lib/logstash/filters/elasticsearch.rb

```diff
@@ -213,6 +217,26 @@ def hosts_default?(hosts)
     hosts.is_a?(Array) && hosts.size == 1 && hosts.first.equal?(DEFAULT_HOST)
   end
 
+  def validate_authentication
```

Comment:
I wouldn't expect a method called `validate_authentication` to _also_ have a side-effect of decoding `@cloud_auth` to ivars and params.

can we keep the original implementation of `fill_user_password_from_cloud_auth` separate and have this method merely perform validation?

Reply:
sure we could, at the same time it is part of the authentication validation. I used the same strategy as the ES output and also in the ES input. I'd like to keep these aligned, it will make it that much easier when working on the integration plugin I guess.

Reply:
+1 to consistency. Ok with resolving here or in a separate effort.

And yeah, these will get a LOT easier when we unify the es plugins 😂

Reply:
🤷‍♂️

Reply:
I put back `fill_user_password_from_cloud_auth`, will do the same in the ES input, for the ES output we can open a PR for that.